### PR TITLE
Updated Image Augmentation tutorial to use Gluon Transforms.

### PR DESCRIPTION
**Pull request by thomelane**
## Description ##
Data Augmentation tutorial used Augmenters rather than Gluon Transforms. Updates have to been made to the tutorial to use Transforms, and new images have been hosted on web-data repository.

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

**Comment by vandanavk** [comment]
@mxnet-label-bot add [Doc, pr-awaiting-review]

File: docs/tutorials/gluon/data_augmentation.md [hunk 1]
```diff
@@ -15,102 +15,220 @@
...
+
+```python
+apply(example_image, transforms.RandomLighting(alpha=1))
 ```
 
+![png](https://raw.githubusercontent.com/dmlc/web-data/master/mxnet/doc/tutorials/gluon/transforms/output_38_0.png)
+
+# Composed Augmentations
+
+In practice, we apply multiple augmentation techniques to an image to increase the variety of images in the dataset. Using the `Compose` transform that was introduced in the [Data Transforms tutorial](), we can apply 3 of the transforms we previously used above.
```
Comment (vishaalkapoor) [inline docs/tutorials/gluon/data_augmentation.md]: the tutorial isn't linked

**Comment by vishaalkapoor** [comment]
Pending changes, LGTM.

File: docs/tutorials/gluon/data_augmentation.md [hunk 2]
```diff
@@ -15,102 +15,220 @@
 <!--- specific language governing permissions and limitations -->
 <!--- under the License. -->
 
-# Methods of applying data augmentation (Gluon API)
+# Image Augmentation
 
-Data Augmentation is a regularization technique that's used to avoid overfitting when training Machine Learning models. Although the technique can be applied in a variety of domains, it's very common in Computer Vision. Adjustments are made to the original images in the training dataset before being used in training. Some example adjustments include translating, cropping, scaling, rotating, changing brightness and contrast. We do this to reduce the dependence of the model on spurious characteristics; e.g. training data may only contain faces that fill 1/4 of the image, so the model trained without data augmentation might unhelpfully learn that faces can only be of this size.
+Augmentation is the process of randomly adjusting samples of your dataset to create new samples that can also be used for neural network training. It increases the variety of samples seen during training and this helps the network avoid overfitting and using spurious characteristics of the dataset.
```
[the remark this answers is not in the document]
Comment (thomelane) [reply docs/tutorials/gluon/data_augmentation.md]: rephrased this whole section.

File: docs/tutorials/gluon/data_augmentation.md [hunk 3]
```diff
@@ -15,102 +15,220 @@
...
+
+Although the technique can be applied in a variety of domains, it's very common in Computer Vision, and we will focus on image augmentations in this tutorial. Some example image augmentations include random crops and flips, and adjustments to the brightness and contrast.
```
[the remark this answers is not in the document]
Comment (thomelane) [reply docs/tutorials/gluon/data_augmentation.md]: think this is covered by my rephrasing.

File: docs/tutorials/gluon/data_augmentation.md [hunk 4]
```diff
@@ -15,102 +15,220 @@
...
+
+#### What are the prerequisites?
+
+You should be familiar with the concept of a transform and how to apply it to a dataset before reading this tutorial. Check out the [Data Transforms tutorial]() if this is new to you or you need a quick refresher.
```
[the remark this answers is not in the document]
Comment (thomelane) [reply docs/tutorials/gluon/data_augmentation.md]: don't have one yet.

File: docs/tutorials/gluon/data_augmentation.md [hunk 5]
```diff
@@ -15,102 +15,220 @@
...
+        for j in range(num_cols):
+            axes[i][j].imshow(imgs[i * num_cols + j].asnumpy())
+            axes[i][j].axes.get_xaxis().set_visible(False)
+            axes[i][j].axes.get_yaxis().set_visible(False)
+    plt.subplots_adjust(hspace=0.1, wspace=0)
+    #print(aspect_ratio)
+    return axes
+
+def apply(img, aug, num_rows=2, num_cols=4, scale=3):
+    Y = [aug(img) for _ in range(num_rows * num_cols)]
```
[the remark this answers is not in the document]
Comment (thomelane) [reply docs/tutorials/gluon/data_augmentation.md]: done.

File: docs/tutorials/gluon/data_augmentation.md [hunk 6]
```diff
@@ -15,102 +15,220 @@
...
+    aspect_ratio = imgs[0].shape[0]/imgs[0].shape[1]
+    figsize = (num_cols * scale, num_rows * scale * aspect_ratio)
+    _, axes = plt.subplots(num_rows, num_cols, figsize=figsize)
+    for i in range(num_rows):
+        for j in range(num_cols):
+            axes[i][j].imshow(imgs[i * num_cols + j].asnumpy())
+            axes[i][j].axes.get_xaxis().set_visible(False)
+            axes[i][j].axes.get_yaxis().set_visible(False)
+    plt.subplots_adjust(hspace=0.1, wspace=0)
+    #print(aspect_ratio)
```
[the remark this answers is not in the document]
Comment (thomelane) [reply docs/tutorials/gluon/data_augmentation.md]: added.

File: docs/tutorials/gluon/data_augmentation.md (hunk 1 above)
Comment (thomelane) [reply docs/tutorials/gluon/data_augmentation.md]: no link yet.

**Comment by thomelane** [comment]
thanks @vishaalkapoor, made changes as suggested.

**Comment by vishaalkapoor** [comment]
LGTM!
